sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -656,25 +643,20 @@ private void setupInitialOffset(PartitionState pState) {
   // Update latest offset for each partition.
   // Called from setupInitialOffset() at the start and then periodically from offsetFetcher thread.
   private void updateLatestOffsets() {
-    for (PartitionState p : partitionStates) {
-      try {
-        Instant fetchTime = Instant.now();
-        consumerSpEL.evaluateSeek2End(offsetConsumer, p.topicPartition);
-        long offset = offsetConsumer.position(p.topicPartition);
-        p.setLatestOffset(offset, fetchTime);
-      } catch (Exception e) {
-        if (closed.get()) { // Ignore the exception if the reader is closed.
-          break;
-        }
-        LOG.warn(
-            "{}: exception while fetching latest offset for partition {}. will be retried.",
-            this,
-            p.topicPartition,
-            e);
-        // Don't update the latest offset.
+    try {
+      final Map<TopicPartition, PartitionState> topicPartitions =
+          partitionStates.stream()
+              .collect(Collectors.toMap(p -> p.topicPartition, Function.identity()));
+      final Map<TopicPartition, Long> newOffsets = consumer.endOffsets(topicPartitions.keySet());
```

Comment:
This runs concurrently with main IO thread (consumerPollLoop()). Kafka consumer is single threaded. Need to move this to IO thread. See how offsets are committed on the main consumer.